homeassistant/components/climate/econet.py

```diff
@@ -0,0 +1,234 @@
+"""
+Support for Rheem EcoNet water heaters.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/climate.econet/
+"""
+import logging
+from os import path
+import datetime
+
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.const import ATTR_ENTITY_ID
+from homeassistant.components.climate import (
+    PLATFORM_SCHEMA,
+    STATE_ECO, STATE_GAS, STATE_ELECTRIC,
+    STATE_HEAT_PUMP, STATE_HIGH_DEMAND,
+    STATE_OFF, STATE_UNKNOWN,
+    ClimateDevice)
+from homeassistant.const import (
+    CONF_PASSWORD, CONF_USERNAME, TEMP_FAHRENHEIT,
+    ATTR_TEMPERATURE)
+from homeassistant.config import load_yaml_config_file
+
+REQUIREMENTS = ['pyeconet==0.0.3']
+
+_LOGGER = logging.getLogger(__name__)
+
+ATTR_VACATION_START = 'next_vacation_start_date'
+ATTR_VACATION_END = 'next_vacation_end_date'
+ATTR_ON_VACATION = 'on_vacation'
+ATTR_TODAYS_ENERGY_USAGE = 'todays_energy_usage'
+ATTR_IN_USE = 'in_use'
+
+ATTR_START_DATE = 'start_date'
+ATTR_END_DATE = 'end_date'
+
+SERVICE_ADD_VACATION = 'econet_add_vacation'
+SERVICE_DELETE_VACATION = 'econet_delete_vacation'
+
+ADD_VACATION_SCHEMA = vol.Schema({
+    vol.Optional(ATTR_ENTITY_ID): cv.entity_ids,
+    vol.Optional(ATTR_START_DATE): cv.positive_int,
+    vol.Required(ATTR_END_DATE): cv.positive_int,
+})
+
+DELETE_VACATION_SCHEMA = vol.Schema({
+    vol.Optional(ATTR_ENTITY_ID): cv.entity_ids,
+})
+
+DOMAIN = 'econet'
```

Comment:
This platform belongs to the climate domain, so rename the variable to `ECONET_DATA`.